### PR TITLE
fix: respect formatting.prefix scss formats

### PR DESCRIPTION
**Pull request by jorenbroekema**
_Issue #, if available:_
https://github.com/amzn/style-dictionary/issues/1286

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: __tests__/formats/__snapshots__/scssMaps.test.snap.js [hunk 1]
```diff
@@ -51,3 +51,53 @@ $tokens: (
...
+$foo-size-font-small: 12rem !default;
+$foo-size-font-large: 18rem !default;
+$foo-color-base-red: #ff0000 !default; // comment
+$foo-color-white: #ffffff !default;
+$foo-asset-icon-book: url("data:image/png;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIyNCIgaGVpZ2h0PSIyNCIgdmlld0JveD0iMCAwIDI0IDI0IiBmaWxsPSJub25lIiBzdHJva2U9ImN1cnJlbnRDb2xvciIgc3Ryb2tlLXdpZHRoPSIyIiBzdHJva2UtbGluZWNhcD0icm91bmQiIHN0cm9rZS1saW5lam9pbj0icm91bmQiIGNsYXNzPSJmZWF0aGVyIGZlYXRoZXItYm9vayI+PHBhdGggZD0iTTQgMTkuNUEyLjUgMi41IDAgMCAxIDYuNSAxN0gyMCI+PC9wYXRoPjxwYXRoIGQ9Ik02LjUgMkgyMHYyMEg2LjVBMi41IDIuNSAwIDAgMSA0IDE5LjV2LTE1QTIuNSAyLjUgMCAwIDEgNi41IDJ6Ij48L3BhdGg+PC9zdmc+") !default;
+
+$tokens: (
+  'size': (
+    'font': (
+      'small': $foo-size-font-small,
```
Comment (jorenbroekema) [inline __tests__/formats/__snapshots__/scssMaps.test.snap.js:85]: these were missing the `$foo-` prefix before this change. Meaning the SCSS result is broken (broken refs)

**Comment by jorenbroekema** [comment]
Oops, looks like some unrelated minor docs improvements made it into this branch, if it's okay I guess we can also merge those while we're at it?

File: lib/common/templates/scss/map-deep.template.js [hunk 2]
```diff
@@ -12,14 +12,15 @@
...
+  const { formatting } = options;
   let output = '';
   if (obj === null) {
     output += `''`;
   } else if (typeof obj === 'string') {
     output += `'${obj}'`;
   } else if (Object.hasOwn(obj, `${options.usesDtcg ? '$' : ''}value`)) {
     // if we have found a leaf (a property with a value) append the value
-    output += `$${obj.name}`;
+    output += `${formatting?.prefix ?? '$'}${obj.name}`;
```
Comment (dbanksdesign) [inline lib/common/templates/scss/map-deep.template.js:23]: It looks like we are missing other `formatting` options as well here like `suffix`, should we add those in too?

**Comment by jorenbroekema** [comment]
Done, making sure the formatting options are all respected.

I tried reusing createPropertyFormatter but this wasn't possible due to the nested nature of these formats.
Suffix is already "respected" in the sense that for map-deep, the top part of it does reuse createPropertyFormatter, where suffix is respected. And in the deep-map below it, you wouldn't want the custom suffix to apply to the inner map syntax, both for flat/deep, that would break the SCSS syntax no matter what.

I think these formatting options are really awkward to be honest. Why would you ever want to change the separator if it breaks the output syntax? Same for prefix, this is better done as part of a token name transform, you're likely to break the output if you do it at the level of output property format. Same for suffix, this either breaks the output or you're better off adding a custom transform that adds a suffix to token values imo. Perhaps something to redesign for v5

**Review by dbanksdesign — APPROVED** [review]

**Pull request merged**